spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/DirectedGraph.scala

```diff
@@ -170,6 +170,17 @@ class Node[T](val element: T) extends Serializable {
   }
 
   /**
+   * Remove linkage with another node
+   *  @param node another node
+   *  @return current node
+   */
+  def delete(node: Node[T]): Node[T] = {
+    if (node.prevs.contains(this)) node.prevs.-=(this)
+    if (this.nexts.contains(node)) this.nexts.-=(node)
+    this
+  }
+
+  /**
```

Comment:
it won't. as per https://github.com/intel-analytics/BigDL/pull/964, we need to add a view before linear if inputsize is not equals to outputsize, so we need to remove it vice versa when persisting.